packages/renderer/src/lib/configmaps-secrets/configmap-secret-utils.ts

```diff
@@ -0,0 +1,56 @@
+/**********************************************************************
+ * Copyright (C) 2023 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import type { V1ConfigMap, V1Secret } from '@kubernetes/client-node';
+
+import type { ConfigMapSecretUI } from './ConfigMapSecretUI';
+
+export class ConfigMapSecretUtils {
+  // If it is a secret, then it will have a type property,
+  // if not it's configmap
+  isSecret(storage: V1ConfigMap | V1Secret): storage is V1Secret {
+    return 'type' in storage && storage.type !== 'ConfigMap';
+  }
+
+  // If it is a configMap, then it will **not** have a type property
+  isConfigMap(storage: V1ConfigMap | V1Secret): storage is V1ConfigMap {
+    return !('type' in storage);
```

Comment:
It would probably be better to have a 'positive' check for something unique about config map vs 'it's not a secret' 😆. Is there nothing that's only in configmap?

Reply:
actually yes! I'll update it to also check for 'binaryData' too.